sanic/mixins/routes.py

```diff
@@ -831,13 +832,17 @@ def _register_static(
             raise ValueError("Invalid file path string.")
         else:
             file_or_directory = static.file_or_directory
-
-        uri = static.uri
-        name = static.name
+        
+        uri = static.uri 
+        name = static.name        
         # If we're not trying to match a file directly,
-        # serve from the folder
-        if not path.isfile(file_or_directory):
+        # serve from the folder        
+        if not static.src and not path.isfile(file_or_directory):
+            uri += "/<__file_uri__:path>"
+        elif static.src == "dir": 
             uri += "/<__file_uri__:path>"
+        elif static.src != "file":
```

Comment:
What if `src` is `None` and it is a file (so the first check is not triggered?)

Reply:
Are you sure about that? We need to add some unit testing. Add some unit tests for all the possible combinations to make sure that the code behaves as expected. Feel free to use an existing test as a template.